doc/lang.md

```diff
@@ -495,59 +487,49 @@ module Foo {
 In our example, module `Foo` has four names in its namespace: `N`, `x`,
 `Init`, and `Next`.
 
-A nested module defines its own namespace, which extends the namespace of the
-outer module:
+Quint does not allow for nested modules. However, you can mimick nested
+namespaces by using `::`. For example:
 
-```scala
-module Outer {
+```bluespec
+module Top {
   var x: int
 
   def xPlus(z) = x + z
 
-  module Inner {
-    val x2 = x + x
-  }
+  val Inner::x2 = x + x
 
-  val inv =
-    (Inner.x2 - x == x)
+  val inv = (Inner::x2 - x == x)
 }
 ```
 
-In the above example, the module `Outer` has the following names in its name
-space: `x`, `xPlus`, `inv`, and `Inner`, whereas the module `Inner` contains
-the names `x2` and the names defined in its parent modules: `x`, `xPlus`, and
-`inv`. The module `Inner` is *stateless*, so the module `Outer` may access the
-definitions of `Inner` via the dot-notation, e.g., `Inner.x2`.
-
-**WARNING:** We are redesigning nested modules,
-see [#496](https://github.com/informalsystems/quint/discussions/496). If you are
-starting with the language, wait a bit before using nested modules.
+Note that in the above example, `Inner::x2` is treated as a qualified identifier, that is, it consists of the prefix `Inner` and the name `x2`.
 
```

Comment:
If that's the case, why not have a `namespace {}` primitive?

Reply:
The idea is obviously stolen from C++ `namespace`. We just thought that we maybe don't need the keyword yet, as one can simply add `Inner::` as a prefix. But we may introduce this keyword one day.